src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -196,86 +190,47 @@
+                                                    :outputs (:invalidOutputs loaded-config)}
+                                   :data (:autocomplete-list @state)}])]))
    :-begin-editing
-   (fn [{:keys [props state locals]}]
-     (when-not (:entities-loaded? @locals)
-       (swap! locals assoc :entities-loaded? true)
-       (endpoints/call-ajax-orch
-        {:endpoint (endpoints/get-entity-types (:workspace-id props))
-         :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                    (if success?
-                      (let [loaded-config (:loaded-config @state)
-                            entity-types (get-parsed-response)
-                            workspace-attributes (get-in props [:workspace :workspace :workspace-attributes])
-                            selected-entity-type (get-in loaded-config [:methodConfiguration :rootEntityType])]
-                        (refresh-autocomplete {:engine (:engine @locals)
-                                               :workspace-attributes workspace-attributes
-                                               :entity-types entity-types
-                                               :selected-entity-type selected-entity-type})
-                        (swap! state assoc :entity-types entity-types))
-                      ;; FIXME: :data-attribute-load-error is unused
-                      (swap! state assoc :data-attribute-load-error status-text)))}))
-     (let [{:keys [loaded-config inputs-outputs]} @state]
-       (swap! state assoc :editing? true :original-config loaded-config :original-inputs-outputs inputs-outputs)))
+   (fn [{:keys [props state locals refs this]}]
+     (if-not (:entities-loaded? @locals)
+       (do (swap! state assoc :blocker "Loading attributes...")
+           (endpoints/call-ajax-orch
+            {:endpoint (endpoints/get-entity-types (:workspace-id props))
+             :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                        (if success?
+                          (let [loaded-config (:loaded-config @state)
+                                entity-types (get-parsed-response)
+                                workspace-attributes (get-in props [:workspace :workspace :workspace-attributes])
+                                selected-entity-type (get-in loaded-config [:methodConfiguration :rootEntityType])]
+                            (swap! locals assoc :entities-loaded? true)
+                            (swap! state assoc
+                                   :blocker nil
+                                   :entity-types entity-types
+                                   :autocomplete-list (build-autocomplete-list
+                                                       (utils/restructure workspace-attributes entity-types selected-entity-type)))
+                            (this :-begin-editing))
+                          ;; FIXME: :data-attribute-load-error is unused
```

Comment:
We should probably fix this at some point. Do we need a new story for it?

Reply:
Yeah, I'll file it.

Reply:
Filed as GAWB-2544